app/src/main/java/com/android/unio/ui/user/UserClaimAssociationPresidentialRights.kt

```diff
@@ -0,0 +1,376 @@
+                                      val code = e.code
+                                      val details = e.details
+                                      Log.e(
+                                          "CloudFunctionError",
+                                          "Error Code: $code, Details: $details",
+                                          e)
+                                    } else {
+                                      Log.e(
+                                          "CloudFunctionError",
+                                          context.getString(
+                                              R.string
+                                                  .user_claim_association_presidential_rights_unexpected_error),
+                                          e)
+                                    }
+                                  } else {
+                                    Log.d("CloudFunction", "OK")
+                                  }
+                                }
+                          }
+                        } else {
+                          // email does not match principalEmailAddress
+                          showErrorMessage = true
+                        }
+                      } else {
+                        showErrorMessage = true
+                        Log.e("UserError", "User does not exist or has no email")
+                      }
+                    } else {
+                      // association is null
+                      showErrorMessage = true
+                      Log.e(
+                          "AssociationError",
+                          "Association does not exist or has no principalEmailAddress")
+                    }
+                  },
+                  modifier =
+                      Modifier.padding(vertical = 8.dp)
+                          .testTag(
+                              UserClaimAssociationPresidentialRightsTestTags.VERIFY_EMAIL_BUTTON)) {
+                    Text("Verify Email")
+                  }
+            } else {
+              // Step 2 ->>> If email is verified, ask for the verification code
+              Text("Enter the code sent to $email:", style = AppTypography.bodySmall)
+              TextField(
+                  value = verificationCode,
+                  onValueChange = { verificationCode = it },
+                  placeholder = {
+                    Text(
+                        context.getString(
+                            R.string
+                                .user_claim_association_presidential_rights_enter_verification_code))
+                  },
+                  modifier =
+                      Modifier.padding(vertical = 8.dp)
+                          .testTag(UserClaimAssociationPresidentialRightsTestTags.CODE))
+              Spacer(modifier = Modifier.height(8.dp))
+
+              val coroutineScope = rememberCoroutineScope()
+
+              Button(
+                  onClick = {
```

Comment:
Great error handling in this function! However, consider extracting it in more modular sub-functions, to decrease mental overload.

Reply:
will refactor that !